File: Problems/0001.-0100./0026._Remove_Duplicates_from_Sorted_Array/RemoveDuplicatesFromSortedArray0.py
#!/bin/python3

class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        i = 0
        while (0 <= i < len(nums)):
            if nums.count(nums[i]) > 1:
                nums.remove(nums[i])
                i -= 1
            i += 1
        return len(nums)

File: Problems/0001.-0100./0026._Remove_Duplicates_from_Sorted_Array/RemoveDuplicatesFromSortedArray0.java
//!/bin/java
class Solution {
    public int removeDuplicates(int[] nums) {
        int i, len = nums.length;
        
        for (i = 0; i < len-1; ++i) {
            if (nums[i] == nums[i+1]) { // Check that no 2 elements are equal; if 
                // they're then there's a duplicate
                int j = i;
                while (j < len-1) {
                    nums[j] = nums[j+1];
                    ++j;
                }
                len--; i--; // decrease the overall length
            }
        }
        return len;
    }
}
